llvm/lib/Target/AMDGPU/SIInstrInfo.h

```diff
@@ -440,6 +440,8 @@ class SIInstrInfo final : public AMDGPUGenInstrInfo {
   bool foldImmediate(MachineInstr &UseMI, MachineInstr &DefMI, Register Reg,
                      MachineRegisterInfo *MRI) const final;
 
+  unsigned getMemOperandAACheckLimit() const override { return 32; }
```

Comment:
Should this be limited based on the subtarget?

Reply:
Good point. Doing that, I won't need upstream